Classes/helpers/class.tx_caretaker_LatestVersionsHelper.php

```diff
@@ -75,10 +77,40 @@ public static function updateLatestTypo3VersionRegistry()
             if (is_array($details) && !empty($details['stable'])) {
                 $stable[$major] = $details['stable'];
             }
+            if (is_array($details) && is_array($details['releases'])) {
+			    $found = false;
+			    $latestCheckedVersion = '';
+			    foreach ($details['releases'] as $version => $versionDetails) {
+			        $latestCheckedVersion = $version;
+                    // if major version > latest_lts check if bugfix part of version number == '0';
+                    // in that case this is the security update and stop searching
+                    if (version_compare($major, $latestLts, '>')) {
+                        if ($versionDetails['type'] === 'security') {
+                            $security[$major] = $version;
+                            $found = true;
+                            break;
+                        }
+                        // if versionDetails->type === 'security' then this is the security update; stop searching
+                        if (self::extractBugfixNumberFromVersion($version) === 0) {
+                            $security[$major] = $version;
+                            $found = true;
+                            break;
+                        }
+                        // if versionDetails->type === 'security' then this is the security update; stop searching
+                    } elseif ($versionDetails['type'] === 'security') {
+                        $security[$major] = $version;
+                        $found = true;
+                        break;
+                    }
+                }
+                if (!$found) {
+			        $security[$major] = $latestCheckedVersion;
+                }
+            }
         }
         \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Registry')->set('tx_caretaker', 'TYPO3versions', $max);
         \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Registry')->set('tx_caretaker', 'TYPO3versionsStable', $stable);
-
+		\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Registry')->set('tx_caretaker', 'TYPO3versionsSecurity', $security);
```

Comment:
Please split these three lines. First get the registry object, then use it to store all three information.